skeleton/common/src/Common/Route/Checked.hs

```diff
@@ -0,0 +1,20 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeApplications #-}
+
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+
+module Common.Route.Checked where
+
+import Data.Functor.Identity
+import qualified Data.Text as Text
+import Obelisk.Route
+
+import Common.Route
+
+case fullRouteEncoder of
+  Left err -> error $ Text.unpack err
+  Right _ ->
+    [d|
+      validFullEncoder :: Encoder Identity Identity (R (FullRoute BackendRoute FrontendRoute)) PageName
+      Right validFullEncoder = fullRouteEncoder
+    |]
```

Comment:
Can't do this directly in `Common.Route` due to stage restriction.
I would also like to have a ```
touchOnlyThisBinding = fullRouteEncoder``` before the case so that renames to `fullRouteEncoder` in "userland" only require a single rename in this weird place (and thus unlikely to get out of sync) but that requires yet another module due to stage restriction.

Reply:
Wasn't able to pull this into a DRY util as there's no way to get a `Lift` instance for `Encoder` to splice back in.
At most we could have a util taking in `Encoder check parse a b` for the check and `Name`/`Dec` for use in the new binding and pass both "forms" of `fullRouteEncoder`. Maybe it'd be worth putting that in `Obelisk.Route` ?